subprojects/core/src/main/java/org/gradle/api/internal/plugins/ApplyPluginBuildOperation.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.plugins;
+
+import org.gradle.api.Action;
+import org.gradle.internal.operations.BuildOperationContext;
+import org.gradle.internal.progress.BuildOperationDetails;
+import org.gradle.internal.progress.BuildOperationExecutor;
+
+public class ApplyPluginBuildOperation {
```

Comment:
The motivation was to put all build operations into a similar pattern to eventually derive an interface from that where everywhere we use a build operation, we essentially  have a `XYBuildOperation.run()` (and XYBuildOperation encapsulates everything else)

But from a performance perspective, we should inline it. Changed it: https://github.com/gradle/gradle/pull/1033/commits/ae801dbd06ec256243a0f2943def5724f7a6f1f9